File: algorithms/AnagramChecker/anagram_checker.py
def to_counts(s):
    counts = {}
    for c in s:
        if c in counts:
            counts[c] += 1
        else:
            counts[c] = 1
    return counts


def is_anagram(s1, s2):
    return to_counts(s1) == to_counts(s2)


if __name__ == "__main__":
    s1 = input("Enter a word: ")
    s2 = input("Enter another word: ")

    if is_anagram(s1, s2):
        print(s1 + " and " + s2 + " are anagrams")
    else:
        print(s1 + " and " + s2 + " are not anagrams")

File: algorithms/AnagramChecker/AnagramChecker.kt
fun toSortedArray(str: String): List<Char> {
    val chars = str.lowercase().toCharArray()
    return chars.sortedArray().filter{ it != ' ' }
}

fun isAnagram(s1: String, s2: String): Boolean {
    return toSortedArray(s1).equals(toSortedArray(s2))
}

fun main() {
    print("Enter a word: ")
    val s1 = readln()
    print("Enter another word: ")
    val s2 = readln()

    if (isAnagram(s1, s2)) {
        println("$s1 and $s2 are anagrams.")
    } else {
        println("$s1 and $s2 are not anagrams.")
    }
}
